src/agent/onefuzz/src/syncdir.rs

```diff
@@ -20,35 +27,35 @@ pub enum SyncOperation {
 const DELAY: Duration = Duration::from_secs(10);
 const DEFAULT_CONTINUOUS_SYNC_DELAY_SECONDS: u64 = 60;
 
-#[derive(Debug, Deserialize, Clone, PartialEq, Default)]
+#[derive(Debug, Deserialize, Clone, PartialEq)]
 pub struct SyncedDir {
     pub path: PathBuf,
-    pub url: Option<BlobContainerUrl>,
+    pub url: BlobContainerUrl,
 }
 
 impl SyncedDir {
     pub async fn sync(&self, operation: SyncOperation, delete_dst: bool) -> Result<()> {
-        if self.url.is_none() {
-            debug!("not syncing as SyncedDir is missing remote URL");
-            return Ok(());
-        }
-
-        let dir = &self.path;
-        let url = self.url.as_ref().unwrap().url();
-        let url = url.as_ref();
-        debug!("syncing {:?} {}", operation, dir.display());
-        match operation {
-            SyncOperation::Push => az_copy::sync(dir, url, delete_dst).await,
-            SyncOperation::Pull => az_copy::sync(url, dir, delete_dst).await,
+        // Adding a trailing '/' to specify a copy of the dir content on linux
```

Comment:
Wow, I didn't even know we could do this in this way.

If this is necessary, maybe it should be the responsibility of the `az_copy` module?

Reply:
moved the logic into the SyncPath type